Orange/widgets/visualize/owscatterplot.py

```diff
@@ -252,6 +252,8 @@ def set_data(self, data):
         self.vizrank_button.setEnabled(
             self.data is not None and not self.data.is_sparse() and
             self.data.domain.class_var is not None and
+            (len(self.data.domain.class_var.values)
+             if self.data.domain.class_var.is_discrete else True) and
```

Comment:
This fixes the case when the class has no values. How about when the class has e.g. 3 values, but all instances have unknown classes (e.g. a test set with the same variable as the training set, but class unknown)